app/controllers/inventories_controller.rb

```diff
@@ -0,0 +1,41 @@
+class InventoriesController < ApplicationController
+  skip_before_action :authenticate_user!, only: [:index, :search]
+
+  def index
+    @inventories = Inventory.where(user_id: current_user.id)
+    @admin = admin?
+
+  end
+
+  def show
+    @inventory = Inventory.find(params[:id])
+    @admin = admin?
+  end
+
+  def update
+    @inventory = Inventory.find(params[:id])
+    @inventory.update(inventory_params)
+    @inventory.save
+    redirect_to market_path
+  end
+
+  def destroy
+    @inventory = Inventory.find(params[:id])
+    @inventory.destroy
```

Comment:
@obj.destroy if @obj.present? like in QuestionController and do it in ItemsController

Reply:
Deleted like useless method